src/registry/v1/data/device/mod.rs

```diff
@@ -77,9 +78,9 @@ impl Device {
     /// If there are no credentials already existing an array is created
     /// if there is an error deserializing the existing data an error is returned
     pub fn add_credential(&mut self, credential: Credential) -> Result<(), serde_json::Error> {
-        self.update_section::<DeviceSpecCredentials, _>(|mut credentials| {
-            credentials.credentials.push(credential);
-            credentials
+        self.update_section::<DeviceSpecAuthentication, _>(|mut auth| {
```

Comment:
This kinda breaks the compatibility silently. 
drg uses that to add a credential to a device object expecting it to work on drogue-cloud, while drogue cloud will call `device.section::<DeviceSpecCredentials>();` and retrieve `None`

Reply:
Yes, but not until drg updates to the newer drogue-client, which I intend to address in a PR to drg.

For drogue-cloud, the plan is to have it look at both .authentication first and fallback to .credentials when looking up information, in order to maintain backwards compatibility with older drg versions.

Do we expect newer drg versions to be backwards compatible with older versions of drogue cloud as well? If that's the case, we need to agree on when how many versions we support the 'old' format in drg.

Reply:
@jbtrystram I added backwards compat support now, and a TODO to remove it after 2 next releases.

Reply:
I think it is good to take this case as a guinea pig for API/data format changes. But I also wouldn't spend too much time on it.

Ideally I guess we should: 

* let the authentication service read the old format too
* convert to the new format at least when storing an update, maybe also when reading it

The latter point goes into the direction of validating configuration in the device registry. Which we also considered: drogue-iot/drogue-cloud#264

If we would have the latter, then a newer version of drg would automatically be able to read/write the data.

Maybe it would also be worth adding some kind of apiVersion to the resource?!